pkg/bzimage/bzImage.go

```diff
@@ -46,15 +47,26 @@ func (b *BzImage) UnmarshalBinary(d []byte) error {
 	b.Kernel = d[b.KernelOffset:]
 	Debug("Kernel at %d, %d bytes", b.KernelOffset, len(b.Kernel))
 	b.KernelBase = uintptr(0x100000)
-	b.InitRAMFS = d[b.Header.RamDiskImage : b.Header.RamDiskImage+b.Header.RamDiskSize]
+	if b.Header.RamDiskImage == 0 {
```

Comment:
I feel this would make more sense if you invert the if-statements, so:

```
	if b.Header.RamDiskImage != 0 {
		i := b.Header.RamDiskImage - uint32(b.KernelBase)
		s := b.Header.RamDiskSize
		Debug("initrd in file at %d, %d bytes", i, s)
		b.InitRAMFS = d[i : i+s]
		Debug("Ramdisk at %d, %d bytes", b.Header.RamDiskImage, b.Header.RamDiskSize)
	}
	return nil
```

Reply:
I see your point but I'm still inclined to leave it for now? I'm never sure about this kind of case.